### PR TITLE
improve speed of caldav event fetching

### PR DESCRIPTION
**Pull request by femyeda**
improve speed of caldav event fetching by properly passing a timeRange instead of fetching entire calendar history

uses forked version of tsDav
https://github.com/calendso/tsDAV

### CODE REVIEW

**Comment by femyeda** [comment]
I see that https://github.com/llldar/tsDAV has updated the parsing of the dates; however, it does not look to be thorough enough.

**Comment by emrysal** [comment]
I think we should pick this up with https://github.com/llldar/tsDAV - as the library is still active.

```javascript
start: timeRange?.start.replace(/[-:.]/g, ''),
end: timeRange?.end.replace(/[-:.]/g, ''),

// and for sure:
const timezone = rs.props?.calendarTimezon;
```

**Comment by emrysal** [comment]
Having looked into this it looks to me that the time format that is expected is `20181017T204900Z` - if I apply the `timeRange?.end.replace(/[-:.]/g, '')` on a typical time format: 2018-10-17T20:49:00Z - this is thorough enough according to https://datatracker.ietf.org/doc/html/rfc5545.

Our fork however passes the raw data assuming it is passed correctly; but after having explored the source for `${new ICAL.Time.fromJSDate(new Date(dateFrom), true).toICALString()}` (ICAL.Time() toString proxy) I think it just generates a full ISO 8601 string with seperators. the RFC leaves no doubt that any implementation is allowed not to use seperators, so even though this may pass in your clients it may not elsewhere.

```javascript
toString: function toString() {
      var result = this.year + '-' +
                   ICAL.helpers.pad2(this.month) + '-' +
                   ICAL.helpers.pad2(this.day);

      if (!this.isDate) {
          result += 'T' + ICAL.helpers.pad2(this.hour) + ':' +
                    ICAL.helpers.pad2(this.minute) + ':' +
                    ICAL.helpers.pad2(this.second);

        if (this.zone === ICAL.Timezone.utcTimezone) {
          result += 'Z';
        }
      }

      return result;
    },
```

This makes me confused why we don't just use:
```javascript
// ISO 8601 with seperators like `2018-10-17T20:49:00Z`
// note this MAY generate invalid `2018-10-17T20:49:00+00:00` 
// ISO 8601 is safer with `dayjs(...).format("YYYY-MM-DDTHH:MM:SS[Z]")`.
+ dayjs(dateFrom).format()
- ${new ICAL.Time.fromJSDate(new Date(dateFrom), true).toICALString()} // output format: YYYY-MM-DDTHH:MM:SS[Z]

// or, as the toICALString() include seperator characters, here's dayjs() without:
+ dayjs(dateFrom).format("YYYYMMDDTHHMMSS[Z]")
```

**Comment by femyeda** [comment]
>but after having explored the source for ${new ICAL.Time.fromJSDate(new Date(dateFrom), true).toICALString()} (ICAL.Time() toString proxy) I think it just generates a full ISO 8601 string with seperators. 

Actually, toICALString() does not produce ISO 8601 with separators. 

Fork or not, 
```
{
   start: timeRange?.start.replace(/[-:.]/g, ''),
   end: timeRange?.end.replace(/[-:.]/g, ''),
}
```

Previously that code was unambiguous as to which format you needed. And the conversion was "naive". It was something chop off the last 5 characters and add "Z"

The method I choose was the safest. 
The new time conversion in llldar/tsDav expects ISO 8601 so I'm fine with using the fork now.

**Comment by femyeda** [comment]
Will update the package, and pass it ISO 8601 using Dayjs

File: lib/integrations/CalDav/CalDavCalendarAdapter.ts [hunk 1]
```diff
@@ -229,11 +235,13 @@ export class CalDavCalendar implements CalendarApiAdapter {
 
       return calendars
         .filter((calendar) => {
```
Comment (emrysal) [inline lib/integrations/CalDav/CalDavCalendarAdapter.ts:237]: Does this work?
```javascript
return calendar.components?.includes("VEVENT")
```

File: lib/integrations/CalDav/CalDavCalendarAdapter.ts [hunk 2]
```diff
@@ -229,11 +235,13 @@ export class CalDavCalendar implements CalendarApiAdapter {
...
-          return calendar.components.includes("VEVENT");
+          return calendar.components && calendar.components.includes("VEVENT")
```
Comment (emrysal) [inline lib/integrations/CalDav/CalDavCalendarAdapter.ts:238]: I think this evaluates to always be true; I checked in the console to verify [] is also truthy.

```javascript
calendar.components?.includes("VEVENT") // instead maybe?
```

Comment (femyeda) [reply lib/integrations/CalDav/CalDavCalendarAdapter.ts:238]: It's a check on null or undefined, but ?.includes is cleaner regardless.

**Comment by femyeda** [comment]
So even though toICALString produces the correct format, and also works v1.0.6 of llldar/tsDav
```
 start: dayjs(dateFrom).utc(false).format("YYYY-MM-DDTHH:MM:ss[Z]"),
end: dayjs(dateTo).utc(false).format("YYYY-MM-DDTHH:MM:ss[Z]"),
```

Using DayJS to format the date for consistency with other parts of the code.

**Comment by kuba-orlik** [comment]
Just leaving a trace that this is related to #479

**Review by emrysal — APPROVED** [review]

**Pull request merged**